internal/toolexec/proxy/generator/main.go

```diff
@@ -0,0 +1,258 @@
+				}
+				if _, isKnown := knownFlags[flag]; !isKnown {
+					log.Fatalf("Unknown captured flag: %q\n", flag)
+				}
+				capturedFlags[flag] = field.Names[0].Name
+			}
+		}
+	}
+
+	file := jen.NewFile(goPackage)
+	file.HeaderComment("// Unless explicitly stated otherwise all files in this repository are licensed")
+	file.HeaderComment("// under the Apache License Version 2.0.")
+	file.HeaderComment("// This product includes software developed at Datadog (https://www.datadoghq.com/).")
+	file.HeaderComment("// Copyright 2023-present Datadog, Inc.\n")
+	file.HeaderComment("// Code generated by 'go generate' DO NOT EDIT.")
+
+	file.Func().
+		Params(jen.Id("f").Op("*").Id(typeName)).
+		Id("parse").
+		Params(jen.Id("args").Index().String()).
+		Params(
+			jen.Index().String(),
+			jen.Error(),
+		).
+		BlockFunc(func(g *jen.Group) {
+			g.Id("flagSet").
+				Op(":=").
+				Qual("flag", "NewFlagSet").
+				Call(
+					jen.Lit(command),
+					jen.Qual("flag", "ContinueOnError"),
+				)
+
+			reDefault := regexp.MustCompile(`^(.+?)(?:\s+\(default\s+(.+?)\))?$`)
+			for _, spec := range flags {
+				fieldName, captured := capturedFlags[spec.Flag]
+
+				var (
+					funcName     string
+					strDefault   string
+					defaultValue *jen.Statement
+				)
+
+				descr := strings.Join(spec.Descr, " ")
+				if matches := reDefault.FindStringSubmatch(descr); matches[2] != "" {
+					strDefault = matches[2]
+					descr = matches[1]
+				}
+
+				switch spec.Value {
+				case "":
+					funcName = "Bool"
+					if strDefault == "" || strDefault == "false" {
+						defaultValue = jen.False()
+					} else {
+						defaultValue = jen.True()
+					}
+				case "int":
+					funcName = "Int"
+					if strDefault != "" {
+						val, err := strconv.Atoi(strDefault)
+						if err != nil {
+							log.Fatalf("Invalid default value for an int flag: %s\n", strDefault)
+						}
+						defaultValue = jen.Lit(val)
+					} else {
+						defaultValue = jen.Lit(0)
+					}
+				default:
+					funcName = "String"
+					defaultValue = jen.Lit(strDefault)
+				}
+
+				flagName := jen.Lit(spec.Flag[1:])
+
+				if funcName == "Bool" && spec.Flag == "-V" {
+					var handler *jen.Statement
+					if captured {
+						handler = jen.Func().Params(jen.String()).Error().Block(
+							jen.Id("f").Dot(fieldName).Op("=").True(),
+							jen.Return(jen.Nil()),
+						)
+					} else {
+						handler = jen.Func().Params(jen.String()).Error().Block(jen.Return(jen.Nil()))
+					}
+					g.Id("flagSet").Dot("BoolFunc").Call(flagName, jen.Lit(descr), handler)
+				} else if captured {
+					g.Id("flagSet").Dot(funcName+"Var").Call(jen.Op("&").Id("f").Dot(fieldName), flagName, defaultValue, jen.Lit(descr))
+				} else {
+					g.Id("flagSet").Dot(funcName).Call(flagName, defaultValue, jen.Lit(descr))
+				}
+			}
+			g.Line()
+
+			g.Id("err").Op(":=").Id("flagSet").Dot("Parse").Call(jen.Id("args"))
+			g.Return(
+				jen.Id("flagSet").Dot("Args").Call(),
+				jen.Id("err"),
+			)
+		})
+
```

Comment:
```suggestion
		if len(capturedFlags) > 0 {
			log.Fatalf("Unknown flags used as 'ddflag' tag in struct: %q\n", capturedFlags)
		}
```

Reply:
This is already checked for when the capture tags are processed.